corehq/messaging/smsbackends/infobip/models.py

```diff
@@ -0,0 +1,66 @@
+from corehq.apps.sms.models import SQLSMSBackend, SMS
+from corehq.apps.sms.util import clean_phone_number
+from corehq.messaging.smsbackends.infobip.forms import InfobipBackendForm
+
+
+class SQLInfobipBackend(SQLSMSBackend):
+
+    class Meta(object):
+        app_label = 'sms'
+        proxy = True
+
+    @classmethod
+    def get_available_extra_fields(cls):
+        return [
+            'account_sid',
+            'auth_token',
+            'scenario_key'
+        ]
+
+    @classmethod
+    def get_api_id(cls):
+        return 'INFOBIP'
+
+    @classmethod
+    def get_generic_name(cls):
+        return "Infobip"
+
+    @classmethod
+    def get_form_class(cls):
+        return InfobipBackendForm
+
+    @classmethod
+    def get_opt_in_keywords(cls):
+        return ['START']
+
+    @classmethod
+    def get_pass_through_opt_in_keywords(cls):
+        return ['YES']
+
+    @classmethod
+    def get_opt_out_keywords(cls):
+        return ['STOP', 'STOPALL', 'UNSUBSCRIBE', 'CANCEL', 'END', 'QUIT']
+
+    def send(self, msg, orig_phone_number=None, *args, **kwargs):
+        config = self.config
+        to = clean_phone_number(msg.phone_number)
+        try:
+            self._send_text_message(config, to, msg)
```

Comment:
How does this handle WhatsApp templates? From what I could gather from a quick read of the InfoBip API docs, you need to pass through all the whatsapp template info to send a template message (i.e. `templateData`, `templateNamespace`, etc.)
I'm guessing you already saw the Turn integration we have (since this looks like it was copy/pasted from there) which implements this functionality.

Reply:
You are right, thanks for the heads-up. I'll return to this section and implement wa templates as soon as full access gets approved for our account. For now leaving a TODO and a comment on the PR.